app/next-client-app/components/scanreports/ScanReportTableUpdateForm.tsx

```diff
@@ -0,0 +1,124 @@
+"use client";
+
+import { Button } from "@/components/ui/button";
+import { updateScanReportTable } from "@/api/scanreports";
+import { Save } from "lucide-react";
+import { toast } from "sonner";
+import { FormDataFilter } from "../form-components/FormikUtils";
+import { Form, Formik } from "formik";
+import { Tooltips } from "../Tooltips";
+import { FormikSelect } from "../form-components/FormikSelect";
+
+interface FormData {
+  personId: number | null;
+  dateEvent: number | null;
+}
+
+export function ScanReportTableUpdateForm({
+  scanreportFields,
+  scanreportTable,
+  permissions,
+  personId,
+  dateEvent,
+}: {
+  scanreportFields: ScanReportField[];
+  scanreportTable: ScanReportTable;
+  permissions: Permission[];
+  personId: ScanReportField;
+  dateEvent: ScanReportField;
+}) {
+  const canUpdate =
+    permissions.includes("CanEdit") || permissions.includes("CanAdmin");
+
+  const fieldOptions = FormDataFilter<ScanReportField>(scanreportFields);
+
+  const initialPersonId = FormDataFilter<ScanReportField>(personId);
+  const initialDateEvent = FormDataFilter<ScanReportField>(dateEvent);
+
+  const handleSubmit = async (data: FormData) => {
+    const submittingData = {
+      person_id: data.personId !== 0 ? data.personId : null,
+      date_event: data.dateEvent !== 0 ? data.dateEvent : null,
+    };
+
+    const response = await updateScanReportTable(
+      scanreportTable.id,
+      submittingData,
+      scanreportTable.scan_report
+    );
+    if (response) {
+      toast.error(
+        `Update Scan Report Table failed. Error: ${response.errorMessage}`
+      );
+    } else {
+      toast.success("Update Scan Report Table successful!");
+    }
+  };
+
+  return (
+    <Formik
+      initialValues={{
+        dateEvent: initialDateEvent[0].value,
+        personId: initialPersonId[0].value,
+      }}
+      onSubmit={(data) => {
+        handleSubmit(data);
```

Comment:
should this not be asynchronous?

Reply:
I think handleSubmit should be an async function, to enable await update SR table to be done, then show the according toast (which is from lines 38-56)

Reply:
I mean since `handleSubmit` is asynchronous, should we not use `async` and `await` in `onSubmit`?

Reply:
I see.
I think because onSubmit doesn't contain any code that needs to wait for handleSubmit to be complete, and I am handling the result of HandleSubmit inside itself (don't need to return a response, then handle this on onSubmit), async and await are not needed on OnSubmit.